bootstrap/bootstrap.go

```diff
@@ -1058,8 +1058,10 @@ func (b *Bootstrap) updateGitMirror() (string, error) {
 		return "", err
 	}
 
-	// Update our mirror
-	if err := b.shell.Run("git", "--git-dir", mirrorDir, "remote", "update", "--prune"); err != nil {
+	// Update our mirror, but just the branch requested. Otherwise on huge repositories this can be
+	// exceedingly slows to run
+	gitFetchFlags := fmt.Sprintf("--git-dir %q %s", mirrorDir, b.GitFetchFlags)
+	if err := gitFetch(b.shell, gitFetchFlags, "origin", b.Branch); err != nil {
```

Comment:
I wonder if we actually want to pass `b.GitFetchFlags` to git-mirror operations.

It was added in https://github.com/buildkite/agent/pull/957 where it seems the use-case was for `--depth=1` to maintain shallow clones. Do we want shallow cloned mirrors? I wonder what else people might be using GitFetchFlags for, and whether they should or shouldn't apply to mirrors.

We have separate `GitCloneFlags` and `GitCloneMirrorFlags` so that config isn't shared. I think if anything we'd need to add `GitFetchMirrorFlags`. But maybe we can get away without user-configurable fetch flags for the mirror?